### ISSUE TITLE
[HW3] redux connect된 component의 setState를 어떻게 수행할 수 있을까요

### ISSUE THREAD

**Issue by jaewooklee93** [swsnu/swppfall2020#149]
제가 사용해본 몇몇 방법들입니다.
```javascript
  it(`should have the following fields in Preview tab`, () => {
    const component = mount(articleCreate);
    const instance = component.find(ArticleCreate.WrappedComponent).instance()
    instance.setState({tab: "PREVIEW"}); // called but no re-rendering
    component.setState({tab: "PREVIEW"}); // fail to call

    expect(component.find("#article-author").length).toBe(1); // should pass if re-rendered
```
처음에 ArticleCreate 페이지에 들어가면 기본설정이 WRITE 탭으로 되어있어서 이를 PREVIEW 탭으로 강제로 `setState`하여 `render`를 다시 한 후 테스트를 해보고 싶었습니다.

물론 preview 버튼을 눌러서 테스트를 할수도 있긴 한데 state를 직접 바꾸는 것이 가능하다면 그게 더 편리할 것 같습니다.

`instance.setState`로 하면 함수 호출에는 문제가 없지만 다시 `render`해주지 않는 것 같습니다. `component.setState`로 하면 호출조차 되지 않습니다. 혹시 어떤 방법을 쓸 수 있을지 도움 요청드립니다.

**Comment by gina7484** [comment]
Preview tab 버튼을 찾아서 click을 simulate한 후에 .instance()를 호출해주면 preview탭을 누른것이 반영된 채로 render가 될 것 같습니다.

**Comment by jaewooklee93** [comment]
@gina7484 네 맞습니다. 버튼을 직접 찾아서 `.simulate('click')` 해주면 일단 green으로 테스트가 통과되기는 합니다. 그렇지만 혹시 `setState`를 .test.js 파일 상에서 직접 수행할 수 있는 방법은 없을지 궁금합니다.

**Comment by hy00nc** [comment]
`setState()`가 항상 re-render을 보장해주지 않기 때문에 re-render가 일어나도록 하고 싶으시면 `forceupdate()`를 쓰는 방법도 있습니다. https://reactjs.org/docs/react-component.html#forceupdate

**Comment by jaewooklee93** [comment]
감사합니다. 말씀하신 대로 해보니 잘 작동하는 것 같습니다
